test/parallel/test-fs-opendir.js

```diff
@@ -188,7 +188,7 @@ async function doAsyncIterThrowTest() {
     }
   }
 
-  await assert.rejects(async () => dir.read(), dirclosedError);
+  await assert.rejects(dir.read(), dirclosedError);
```

Comment:
hmm, the title says `dir.close()` but the tests are looking at `dir.read()` with the `dirclosedError`. Just a bit confusing. Some additional code comments could help